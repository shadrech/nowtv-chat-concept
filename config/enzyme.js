var enzyme = require('enzyme');
var Adapter = require('enzyme-adapter-react-15');

enzyme.configure({ adapter: new Adapter() });