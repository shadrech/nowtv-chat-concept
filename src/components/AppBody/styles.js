import styled from "react-emotion";

export const BodyWrapper = styled('div')`
  background: red;
  display: flex;
  position: relative;
  width: 100%;
  height: calc(100vh - 6.7rem);
`;

export const ChatLogWrapper = styled('div')`
  flex-grow: 3;
  background: #e5ecef;
  padding: 4rem 5rem;
  overflow-y: scroll;
`;