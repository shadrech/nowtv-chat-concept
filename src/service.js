import { getMessages, getMembers } from "./data";

export default async function getChatLog() {
  const promises = [getMessages(), getMembers()];
  const [messages, members] = await Promise.all(promises);

  const chatData = members.reduce((acc, {id, email, avatar, lastName, firstName}) => {
    const memberMessages = messages.filter(({userId}) => userId === id);
    return acc.concat(memberMessages.map(message => ({
      messageId: message.id,
      userId: id,
      fullName: `${firstName} ${lastName}`,
      timestamp: message.timestamp,
      email,
      message: message.message,
      avatar
    })));
  }, []);

  return chatData.sort((c1, c2) => {
    const d1 = new Date(c1.timestamp);
    const d2 = new Date(c2.timestamp);
    return d2 - d1;
  });
};