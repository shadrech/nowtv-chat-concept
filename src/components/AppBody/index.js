import React, {Component} from "react";

import { BodyWrapper, ChatLogWrapper } from "./styles";
import MembersArea from "../MembersArea";
import Message from "../../widgets/Message";

class AppBody extends Component {
  render() {
    const {messages, members, activeMember, ...rest /*loading, error*/} = this.props;
    const mssgs = activeMember ? messages.filter(m => m.userId === activeMember.id) : messages;

    return (
      <BodyWrapper>
        <ChatLogWrapper>
          {mssgs.map((m, i) => <Message message={m} alignment={activeMember || i%2 === 0 ? "right" : "left"} key={m.messageId} />)}
        </ChatLogWrapper>
        <MembersArea members={members} activeMember={activeMember} {...rest} />
      </BodyWrapper>    
    );
  }
}

export default AppBody;