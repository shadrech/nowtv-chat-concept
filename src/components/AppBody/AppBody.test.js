import React from "react";
import {shallow} from "enzyme";

import AppBody from "./";
import { dummyData } from "../../service.test";

describe("<AppBody />", () => {
  const {messages, members} = dummyData;
  test("should render as expected", async () => {
    const appBody = shallow(<AppBody messages={messages} members={members} />);
    expect(appBody).toMatchSnapshot();
    
  })
})