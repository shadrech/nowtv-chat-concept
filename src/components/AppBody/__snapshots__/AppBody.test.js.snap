// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppBody /> should render as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AppBody
    members={
      Array [
        Object {
          "avatar": null,
          "email": "abakera@craigslist.org",
          "firstName": "Amanda",
          "id": "73f30d93-f87b-45ea-bda5-60fa4b3716b5",
          "ip": "82.87.241.19",
          "lastName": "Baker",
          "messageCount": 3,
        },
      ]
    }
    messages={
      Array [
        Object {
          "avatar": "http://dummyimage.com/100x100.bmp/5fa2dd/ffffff",
          "email": "lowensm@earthlink.net",
          "fullName": "Larry Owens",
          "message": "Integer a nibh. In quis justo. Maecenas rhoncus aliquam lacus.",
          "messageId": "b03569ae-ccbf-4975-8040-4daba638b407",
          "timestamp": "2016-11-09T05:04:58Z",
          "userId": "16373df5-da0a-4074-8295-f916b94269f4",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Styled(div)>
          <Message
            alignment="right"
            message={
              Object {
                "avatar": "http://dummyimage.com/100x100.bmp/5fa2dd/ffffff",
                "email": "lowensm@earthlink.net",
                "fullName": "Larry Owens",
                "message": "Integer a nibh. In quis justo. Maecenas rhoncus aliquam lacus.",
                "messageId": "b03569ae-ccbf-4975-8040-4daba638b407",
                "timestamp": "2016-11-09T05:04:58Z",
                "userId": "16373df5-da0a-4074-8295-f916b94269f4",
              }
            }
          />
        </Styled(div)>,
        <MembersArea
          activeMember={undefined}
          members={
            Array [
              Object {
                "avatar": null,
                "email": "abakera@craigslist.org",
                "firstName": "Amanda",
                "id": "73f30d93-f87b-45ea-bda5-60fa4b3716b5",
                "ip": "82.87.241.19",
                "lastName": "Baker",
                "messageCount": 3,
              },
            ]
          }
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Message
              alignment="right"
              message={
                Object {
                  "avatar": "http://dummyimage.com/100x100.bmp/5fa2dd/ffffff",
                  "email": "lowensm@earthlink.net",
                  "fullName": "Larry Owens",
                  "message": "Integer a nibh. In quis justo. Maecenas rhoncus aliquam lacus.",
                  "messageId": "b03569ae-ccbf-4975-8040-4daba638b407",
                  "timestamp": "2016-11-09T05:04:58Z",
                  "userId": "16373df5-da0a-4074-8295-f916b94269f4",
                }
              }
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "b03569ae-ccbf-4975-8040-4daba638b407",
            "nodeType": "function",
            "props": Object {
              "alignment": "right",
              "message": Object {
                "avatar": "http://dummyimage.com/100x100.bmp/5fa2dd/ffffff",
                "email": "lowensm@earthlink.net",
                "fullName": "Larry Owens",
                "message": "Integer a nibh. In quis justo. Maecenas rhoncus aliquam lacus.",
                "messageId": "b03569ae-ccbf-4975-8040-4daba638b407",
                "timestamp": "2016-11-09T05:04:58Z",
                "userId": "16373df5-da0a-4074-8295-f916b94269f4",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeMember": undefined,
          "members": Array [
            Object {
              "avatar": null,
              "email": "abakera@craigslist.org",
              "firstName": "Amanda",
              "id": "73f30d93-f87b-45ea-bda5-60fa4b3716b5",
              "ip": "82.87.241.19",
              "lastName": "Baker",
              "messageCount": 3,
            },
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Styled(div)>
            <Message
              alignment="right"
              message={
                Object {
                  "avatar": "http://dummyimage.com/100x100.bmp/5fa2dd/ffffff",
                  "email": "lowensm@earthlink.net",
                  "fullName": "Larry Owens",
                  "message": "Integer a nibh. In quis justo. Maecenas rhoncus aliquam lacus.",
                  "messageId": "b03569ae-ccbf-4975-8040-4daba638b407",
                  "timestamp": "2016-11-09T05:04:58Z",
                  "userId": "16373df5-da0a-4074-8295-f916b94269f4",
                }
              }
            />
          </Styled(div)>,
          <MembersArea
            activeMember={undefined}
            members={
              Array [
                Object {
                  "avatar": null,
                  "email": "abakera@craigslist.org",
                  "firstName": "Amanda",
                  "id": "73f30d93-f87b-45ea-bda5-60fa4b3716b5",
                  "ip": "82.87.241.19",
                  "lastName": "Baker",
                  "messageCount": 3,
                },
              ]
            }
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Message
                alignment="right"
                message={
                  Object {
                    "avatar": "http://dummyimage.com/100x100.bmp/5fa2dd/ffffff",
                    "email": "lowensm@earthlink.net",
                    "fullName": "Larry Owens",
                    "message": "Integer a nibh. In quis justo. Maecenas rhoncus aliquam lacus.",
                    "messageId": "b03569ae-ccbf-4975-8040-4daba638b407",
                    "timestamp": "2016-11-09T05:04:58Z",
                    "userId": "16373df5-da0a-4074-8295-f916b94269f4",
                  }
                }
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "b03569ae-ccbf-4975-8040-4daba638b407",
              "nodeType": "function",
              "props": Object {
                "alignment": "right",
                "message": Object {
                  "avatar": "http://dummyimage.com/100x100.bmp/5fa2dd/ffffff",
                  "email": "lowensm@earthlink.net",
                  "fullName": "Larry Owens",
                  "message": "Integer a nibh. In quis justo. Maecenas rhoncus aliquam lacus.",
                  "messageId": "b03569ae-ccbf-4975-8040-4daba638b407",
                  "timestamp": "2016-11-09T05:04:58Z",
                  "userId": "16373df5-da0a-4074-8295-f916b94269f4",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeMember": undefined,
            "members": Array [
              Object {
                "avatar": null,
                "email": "abakera@craigslist.org",
                "firstName": "Amanda",
                "id": "73f30d93-f87b-45ea-bda5-60fa4b3716b5",
                "ip": "82.87.241.19",
                "lastName": "Baker",
                "messageCount": 3,
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "prevContext": true,
          },
        },
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
